protocol/auditlog.go

```diff
@@ -11,10 +11,21 @@ import (
 )
 
 type directoryHistory struct {
+	*AudState
 	addr      string
-	signKey   sign.PublicKey
-	snapshots map[uint64]*DirSTR
-	latestSTR *DirSTR
+	snapshots []*DirSTR
```

Comment:
This is a possibility, but the slice makes it much easier to append a whole batch of STRs. I wanted to avoid having to loop through a range twice, once for the verification, and one for the insertion into `snapshots`. But maybe this wouldn't be a huge performance issue?